core/kotlinx-coroutines-core/src/CoroutineContextThreadLocal.kt

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2016-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.coroutines.experimental
+
+import kotlin.coroutines.experimental.*
+
+/**
+ * An extension point to define elements in [CoroutineContext] that are installed into thread local
+ * variables every time the coroutine from the specified context in resumed on a thread.
+ *
+ * Implementations on this interface are looked up via [java.util.ServiceLoader].
+ *
+ * Example usage looks like this:
+ *
+ * ```
+ * // declare custom coroutine context element
+ * class MyElement : AbstractCoroutineContextElement(Key) {
+ *     companion object Key : CoroutineContext.Key<MyElement>
+ *     // some state is kept here
```

Comment:
[nitpick] worth adding some real value here, so it will be perfectly clear for users

Reply:
Updated examples.